app/src/main/java/de/danoeh/antennapod/fragment/NewEpisodesFragment.java

```diff
@@ -97,7 +97,7 @@ public void onHide(FeedItemUndoToken token) {
                     long itemId = token.getFeedItemId();
                     FeedItem item = DBReader.getFeedItem(context, itemId);
                     FeedMedia media = item.getMedia();
-                    if(media != null && media.hasAlmostEnded() && UserPreferences.isAutoDelete()) {
+                    if(media != null && media.hasAlmostEnded() && item.getFeed().getPreferences().getCurrentAutoDelete(UserPreferences.isAutoDelete())) {
```

Comment:
looks rather strange.
What about removing the ``isAutoDelete`` parameter and using 
```
case DEFAULT:
                return UserPreferences.isAutoDelete();
```
?